metrics/open-telemetry-backend/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryBackend.scala

```diff
@@ -0,0 +1,91 @@
+package sttp.client3.opentelemetry
+
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.common.{AttributeKey, Attributes}
+import io.opentelemetry.api.trace.{Span, SpanKind, StatusCode}
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.propagation.{TextMapPropagator, TextMapSetter}
+import sttp.capabilities.Effect
+import sttp.client3._
+import sttp.monad.MonadError
+import sttp.monad.syntax._
+
+import scala.collection.mutable
+
+private class OpenTelemetryBackend[F[_], P](
+    delegate: SttpBackend[F, P],
+    openTelemetry: OpenTelemetry
+) extends SttpBackend[F, P] {
+
+  private val tracer = openTelemetry.getTracer("sttp3-client", "1.0.0")
+  private implicit val _monad: MonadError[F] = responseMonad
+  type PE = P with Effect[F]
+
+  def send[T, R >: PE](request: Request[T, R]): F[Response[T]] = {
+    val carrier: mutable.Map[String, String] = mutable.Map().empty
+    val propagator: TextMapPropagator = openTelemetry.getPropagators.getTextMapPropagator
+    val setter: TextMapSetter[mutable.Map[String, String]] = (carrier, key, value) => carrier.update(key, value)
+    responseMonad
+      .eval {
+        val attributes = prepareBaseAttributes(request)
+        val span: Span = tracer
+          .spanBuilder(s"HTTP ${request.method.method}")
```

Comment:
I think I'd make this configurable (with sane defaults), like other metrics backend are configurable